src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.opensearch.client.Client;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import demo.DemoWorkflowStep;
+
+/**
+ * Generates instances implementing {@link WorkflowStep}.
+ */
+public class WorkflowStepFactory {
+
+    private static WorkflowStepFactory instance = null;
+
+    private final Map<String, WorkflowStep> stepMap = new HashMap<>();
+
+    /**
+     * Create the singleton instance of this class. Throws an {@link IllegalStateException} if already created.
```

Comment:
I'm just trying to think instead of creating singleton of this class, if we can use some extension point from Plugin.java to achieve the singleton behavior.

Reply:
I've added it to createcontents so it is injected as a singleton and available everywhere that is also injected.

The problem is that we haven't yet implemented getActions which injects the transport actions that will use this.

So yes this is temporary and should eventually be replaced by `@Inject` annotation.  I'm trying to not make this PR any bigger than it already is.